lms/djangoapps/discussion/tasks.py

```diff
@@ -90,3 +93,26 @@ def _get_thread_url(context):
         'id': context['thread_id'],
     }
     return urljoin(settings.LMS_ROOT_URL, permalink(thread_content))
+
+
+def _generate_ga_pixel_url(course_key, user_id):
+    # used for analytics
+    query_params = {
+        'v': '1',
+        't': 'event',
+        'ec': 'email',
+        'ea': 'open',
+        'tid': get_value("GOOGLE_ANALYTICS_TRACKING_ID", settings.GOOGLE_ANALYTICS_TRACKING_ID),
+        'uid': user_id,
+        'utm_source': 'discussion_notification_email',
```

Comment:
should this be "cs"?

Reply:
I got this parameter name from https://ga-dev-tools.appspot.com/campaign-url-builder/ .

Reply:
That builder is for URLs that are linking to your site. You'll want to use them on the link embedded in the email (for example).

From my reading of the docs, the measurement protocol will ignore these and instead use the short "cs" versions. That is: cs == utm_source but for the measurement protocol only.

Reply:
Ah, that makes sense. If we're not going to use Sailthru's GA URL formatting capabilties, then we'll need to add those parameters ourselves, right?